### PR TITLE
Solve 1.6

### PR DESCRIPTION
**Pull request by qarux**

### CODE REVIEW

File: 1_concepts/1_6_dispatch/src/main.rs [hunk 1]
```diff
@@ -1,4 +1,146 @@
...
+#[ derive( Debug ) ]
+struct User
+{
+  id : u64,
+  email : Cow< 'static, str >,
+  activated : bool,
+}
+
+#[ derive( Debug ) ]
+struct UserRepositoryStatic< S >
```
Comment (dmvict) [inline 1_concepts/1_6_dispatch/src/main.rs:21]: Please, restrict type here. It is not ergonomic for large codebase but the task is :
> Implement UserRepository type with injectable Storage implementation

File: 1_concepts/1_6_dispatch/src/main.rs [hunk 2]
```diff
@@ -1,4 +1,146 @@
...
+  fn get( &self, id : u64 ) -> Option< &User >
+  {
+    self.storage.get( &id )
+  }
+
+  fn add( &mut self, user : User ) -> Result< (), () >
+  {
+    if self.get( user.id ).is_some()
+    {
+      return Err( () )
```
Comment (dmvict) [inline 1_concepts/1_6_dispatch/src/main.rs:51]: Please, make proper errors. At least it should be static string.

File: 1_concepts/1_6_dispatch/src/main.rs [hunk 3]
```diff
@@ -1,4 +1,146 @@
...
+  ).unwrap();
+  dynamic_repo.update( User
+  {
+    id : 0,
+    email : Cow::from( "example@example.com" ),
+    activated : true
+  }
+  );
+
+  println!( "static: {:?}", static_repo );
```
Comment (dmvict) [inline 1_concepts/1_6_dispatch/src/main.rs:144]: Please, use assertions.

File: 1_concepts/1_6_dispatch/src/main.rs (hunk 1 above)
Comment (qarux) [reply 1_concepts/1_6_dispatch/src/main.rs]: Done

File: 1_concepts/1_6_dispatch/src/main.rs (hunk 2 above)
Comment (qarux) [reply 1_concepts/1_6_dispatch/src/main.rs]: Done

File: 1_concepts/1_6_dispatch/src/main.rs (hunk 3 above)
Comment (qarux) [reply 1_concepts/1_6_dispatch/src/main.rs]: Done

**Comment by dmvict** [comment]
Can be merged.

**Pull request merged**